src/ui/note/mod.rs

```diff
@@ -415,6 +455,36 @@ impl<'a> NoteView<'a> {
     }
 }
 
+fn get_reposted_note_key(ndb: &Ndb, txn: &Transaction, note: &Note) -> Option<NoteKey> {
```

Comment:
you can just return the note here

```suggestion
fn get_reposted_note<'a>(ndb: &Ndb, txn: &'a Transaction, note: &Note) -> Option<Note<'a>> {
```